src/main/scala/org/broadinstitute/dsde/firecloud/service/CookieAuthedService.scala

```diff
@@ -37,16 +40,41 @@ trait CookieAuthedService extends HttpService with PerRequestCreator with FireCl
         cookie("FCtoken") { tokenCookie =>
           mapRequest(r => addCredentials(OAuth2BearerToken(tokenCookie.content)).apply(r)) { requestContext =>
 
-            // check if the user has access to the file. If so, sign and issue a redirect; if not, replay Google's
-            // exception to the user, just as if the user tried to access the Google file directly.
-            val extReq = Get( HttpGoogleServicesDAO.getObjectResourceUrl(bucket, obj.toString) )
-            val pipeline = authHeaders(requestContext) ~> sendReceive
-            pipeline {extReq} map { response =>
-                response.status match {
-                  case OK =>
-                    val redirectUrl = HttpGoogleServicesDAO.getSignedUrl(bucket, obj.toString)
-                    requestContext.redirect(redirectUrl, StatusCodes.TemporaryRedirect)
-                  case _ => requestContext.complete(response)
+            // query Google for the user's info, then check if the user has access to the file.
+            // If the user is valid and has access, sign and issue a redirect; if not, replay Google's
+            // exception to the user, just as if the user tried to access Google directly.
+            val objectReq = Get( HttpGoogleServicesDAO.getObjectResourceUrl(bucket, obj.toString) )
+            val objectPipeline = authHeaders(requestContext) ~> sendReceive
+
+            val userReq = Get( HttpGoogleServicesDAO.profileUrl )
+            val userPipeline = authHeaders(requestContext) ~> sendReceive
+
+            val objectStr = s"gs://$bucket/$obj"
+
+            userPipeline{userReq} map { userResponse =>
```

Comment:
Is the reason for checking for a valid user so we can log invalid access requests? If so, it would be helpful to see that in a comment too.